library/common/network/configurator.cc

```diff
@@ -167,13 +172,35 @@ void Configurator::reportNetworkUsage(envoy_netconf_t configuration_key, bool ne
 
   // If configuration state changed, refresh dns.
   if (configuration_updated) {
-    refreshDns(configuration_key);
+    refreshDns(configuration_key, false);
+  }
+}
+
+void Configurator::onDnsResolutionComplete(
+    const std::string&, const Extensions::Common::DynamicForwardProxy::DnsHostInfoSharedPtr&,
+    Network::DnsResolver::ResolutionStatus status) {
+  if (enable_drain_post_dns_refresh_ && pending_drain_) {
+    pending_drain_ = false;
+    if (status == Network::DnsResolver::ResolutionStatus::Success) {
+      cluster_manager_.drainConnections();
```

Comment:
let's add a log here? Unless there is some log inside of `drainConnections` (maybe using `ENVOY_LOG_EVENT`?)

Reply:
✅